src/web/services/student.service.ts

```diff
@@ -129,10 +133,58 @@ export class StudentService {
    * Loads list of students from a course in CSV format by calling API.
    */
   loadStudentListAsCsv(queryParams: { courseId: string }): Observable<string> {
-    const paramsMap: Record<string, string> = {
+    return this.httpRequestService.get(ResourceEndpoints.COURSE, {
       courseid: queryParams.courseId,
-    };
-    const responseType: string = 'text';
-    return this.httpRequestService.get(ResourceEndpoints.STUDENTS_CSV, paramsMap, responseType);
+      entitytype: 'instructor',
+    }).pipe(mergeMap((course: Course) => {
+      return this.httpRequestService.get(ResourceEndpoints.STUDENTS, {
+        courseid: queryParams.courseId,
+      }).pipe(map((students: Students) => {
+        return this.processStudentsToCsv(course.courseId, course.courseName, students.students);
+      }));
+    }));
+  }
+
+  processStudentsToCsv(courseId: string, courseName: string, students: Student[]): string {
+    const csvRows: string[][] = [];
+    csvRows.push(['Course ID:', StringHelper.removeExtraSpace(courseId)]);
+    csvRows.push(['Course Name:', StringHelper.removeExtraSpace(courseName)]);
+    csvRows.push([]);
+    const hasSection: boolean = students.filter((student: Student) =>
+        student.sectionName !== 'None' && student.sectionName !== '').length > 0;
+    csvRows.push([hasSection ? 'Section' : '', 'Team', 'Full Name', 'Last Name', 'Status', 'Email']);
+    students.sort((a: Student, b: Student) => {
```

Comment:
Although not explicit, this can be viewed as a "table" of some sort. You should use chain call of `TableComparatorService`:
https://github.com/TEAMMATES/teammates/blob/f4c7e6d1bf248ebf18d44214087ac28da300b891/src/web/app/components/question-responses/per-question-view-responses/per-question-view-responses.component.ts#L117-L123